lib/_http_server.js

```diff
@@ -1010,7 +1010,9 @@ function resOnFinish(req, res, socket, state, server) {
     }
   } else if (state.outgoing.length === 0) {
     if (server.keepAliveTimeout && typeof socket.setTimeout === 'function') {
-      socket.setTimeout(server.keepAliveTimeout);
+      // Increase the internal timeout wrt the advertised value to reduce likeliwood of ECONNRESET errors
+      // due to race conditions between the client and server timeout calculation
+      socket.setTimeout(server.keepAliveTimeout + 1000);
```

Comment:
I don't think this server change is necessary.

Reply:
Might be the cause of all the test failures.

Reply:
This fix protects clients that are using the hint timeout as is (current node impl) without adjusting it for network (or cpu load) delays

Reply:
If server.keepAliveTimeout is set to 0 ([never time out](https://nodejs.org/api/http.html#serverkeepalivetimeout)) this will change it to 1 second.

Reply:
If server.keepAliveTimeout == 0 then the if condition line1012 is false. The timeout on the socket is set only if server.keepAliveTimeout != 0

Reply:
You're correct, sorry I missed that.